### DISCUSSION TITLE
content plugins are not processed in smart search result page

### DISCUSSION THREAD

**Discussion by lucado** [joomla/joomla-cms#47179] [2. Cold Features]
### Steps to reproduce the issue
- insert code like {loadposition myposition} in an article
- perform a search of words just before the code

### Expected result
- visualize the module as in an article
or
- code removed
or
- any better solution

### Actual result
- some text and code {loadposition myposition}

### System information (as much as possible)
- joomla 3.9.12 and 4 alpha 12

### Additional comments
- the module could not be visualized anyway when the text is trumcated for limited number of characters
![screen shot 2019-10-21 at 09 22 44](https://issues.joomla.org/uploads/1/0f2ca6403203ba79bf526ae9169e2da3.png)![screen shot 2019-10-21 at 09 23 15](https://issues.joomla.org/uploads/1/0354415aadd7a16272d49ea55028429b.png)

**Comment by alikon** [comment]
This does'nt happen with com_search the `{loadmoduleid 28}` is correctly stripped out from search result, unfortunately com_search will not be shipped with j4

**Comment by mbabker** [comment]
Because of how Smart Search works, content plugins by design don’t fire off on the frontend search view.  Plugins are supposed to hook the indexing process to do any manipulation to the data that is getting saved, and the load module plugin should be hooking this process to remove the placeholder.

**Comment by Quy** [comment]
@Hackwar

**Comment by Hackwar** [comment]
This is not an issue of smart search, but as Michael wrote, of the content plugins excluding themselfs from fireing during indexing. We should review the core plugins, but Smart Search is working as expected. I've not been responding to this issue so far, because I first wanted to review our plugins.
